### ISSUE TITLE
Proton 5.13 Application tried to create a window, but no driver could be loaded

### ISSUE THREAD

**Issue by serebit** [ValveSoftware/Proton#4266]
With Proton 5.13, upon launching any game that I've tested, the following appears in proton logs:
```
err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
err:winediag:nodrv_CreateWindow Make sure that your X server is running and that $DISPLAY is set correctly.
```
Please only reply to this issue with affirmation IF YOU SEE THESE MESSAGES IN PROTON LOGS.

All of these games worked, and still work, with Proton 5.0-9, but do not launch with 5.13-1:
- TESV Skyrim 
- A Hat in Time
- Call of Duty: World at War ([steam-10090.log](https://github.com/ValveSoftware/Proton/files/5388314/steam-10090.log))
- Call of Duty 4: Modern Warfare 
  - 5.13-1: [steam-7940.log](https://github.com/ValveSoftware/Proton/files/5388307/steam-7940.log)
  - 5.0-9: [steam-7940.log](https://github.com/ValveSoftware/Proton/files/5388330/steam-7940.log)
- PC Building Simulator
- Titanfall 2 ([steam-1237970.log](https://github.com/ValveSoftware/Proton/files/5388322/steam-1237970.log))
- DOOM Eternal ([steam-782330.log](https://github.com/ValveSoftware/Proton/files/5388309/steam-782330.log))
- Mirror's Edge
- Hollow Knight
- NieR: Automata
- ULTRAKILL

I thought it might be an issue with my installation, but I deleted the Proton 5.13 directory and validated files, to no effect. Same result.

I've attempted reinstallation of some of these games, and attempted nuking proton prefixes as well, with no success. It's worth noting that I've talked to others with Nvidia graphics cards who are *not* seeing this issue with TESV Skyrim specifically. Most of these fail to launch due to a DXVK error or Vulkan initialization error, seemingly, but DOOM Eternal fails to launch due to a page fault. All of the games fail to preload several dynamic libraries from `/usr/$LIB`.

**My system info**
CPU: AMD Ryzen 7 3700X
GPU: AMD Radeon RX 580 8GB
Distribution: Solus 4.1
Kernel: Linux 5.6.19
Drivers: Mesa 20.1.9, AMDGPU RADV
LLVM: 10.0.1

[Steam System Information report](https://gist.github.com/serebit/059fb719570d82306f56767e60ce08fd)

**Comment by kisak-valve** [comment]
Hello @serebit, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by nutta-git** [comment]
Titanfall 2 doesn't work for me either 
[steam-1237970.log](https://github.com/ValveSoftware/Proton/files/5388366/steam-1237970.log)
https://gist.github.com/nutta-git/f6bbc7095a4fd3a1380d268b5c4d8674

**Comment by serebit** [comment]
Done, @kisak-valve. Also clarified that *none* of the games I've tested have launched with 5.13, so there have been none in my testing that have worked. Felt like that's worth mentioning.

**Comment by CSahajdacny** [comment]
Proton 5.13 doesn't work for me too.
No game want to work with this version... even Proton logs are not created. 

https://gist.github.com/CSahajdacny/07a4c2e835804bb7368673bd8035d8f4

**Comment by nutta-git** [comment]
#1509

**Comment by kisak-valve** [comment]
This issue report is going to quickly become unusable if all Proton 5.13 issues are dumped here. Unless you have a Proton log with something like the following, please report the issues separately (preferably on the compatibility reports for the games you're seeing the issue with):
```
err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
err:winediag:nodrv_CreateWindow Make sure that your X server is running and that $DISPLAY is set correctly.
```

**Comment by nutta-git** [comment]
@kisak-valve will do, sry

**Comment by kisak-valve** [comment]
I suspect this issue might be an odd interaction between pressure vessel and the Solus's Linux Steam Integration. What settings do you have set in Linux Steam Integration and does setting everything there to the off position have an effect?

**Comment by serebit** [comment]
@kisak-valve All testing I conducted was done with Linux Steam Integration disabled. I can try enabling it.

Edit: Just re-enabled LSI and relaunched Steam, then tried DOOM Eternal. Same result.

**Comment by kuunha** [comment]
I have the same problem using Opensuse Tumbleweed

Witcher 3 [steam-292030.log](https://github.com/ValveSoftware/Proton/files/5388522/steam-292030.log)
Cuphead [steam-268910.log](https://github.com/ValveSoftware/Proton/files/5388526/steam-268910.log)

System Information: https://gist.github.com/kturques/d0c69dfcbb0626a2ce0a2d5ba5561b63

**Comment by ghost** [comment]
I had the same problem, I thought it was the reverse prime configuration of my machine but changing to prime offload it still happens, with 5.0.9 doesnt occur, so I noticed that it happens when you have more than one vulkan icd, in my setup igpu amd and nvidia dgpu, proton ignores the VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.json and tries to load any other of the icd's,  so in my case to test I temporarily moved the radeon_icd*.json to another directory and left the nvidia_icd.json and the games started working normally.

**Comment by kisak-valve** [comment]
Hello @qquique, the filesystem layout seen by Proton 5.13 differs from your host system because it's now being run inside of the Steam Linux Runtime - Soldier environment, I think you'd need to manually convert the path to what it would be inside the container environment for now if you want to use `VK_ICD_FILENAMES`. This is unrelated to the opening post.

**Comment by TheDunkShot** [comment]
I have the same issue, no game using Proton 5.13-1 can run what so ever.
Seems to launch games fine _without_ Steam Linux Runtime - Soldier but with no steam integration and pulls up _normal wine menus_.
There seems to be issues with drivers, **/usr/lib/gvfs/libgvfscommon.so** and **/usr/lib/gio/modules/libgvfsdbus.so** (for me at least)
Proton 5.13-1 cannot launch AT ALL, so I can only get a log from running steam in a terminal.
Plus, _.../common/Proton 5.13-1/proton_dist.tar_ doesn't even unpack/extract, manual extraction does nothing either.
> _Call Of Duty: Infinite Warfare_ - [https://gist.github.com/TheDunkShot/6eb9da6398c54fa916e545010c3de6e0](https://gist.github.com/TheDunkShot/6eb9da6398c54fa916e545010c3de6e0)
> _Lethal League Blaze_ - [https://gist.github.com/TheDunkShot/6ecb97c1f0ca828fdd82ee3f461b6bc1](https://gist.github.com/TheDunkShot/6ecb97c1f0ca828fdd82ee3f461b6bc1)
> _Wreckfest_ - [https://gist.github.com/TheDunkShot/dfd6377e2d8969dc9f93af297f5243bd](https://gist.github.com/TheDunkShot/dfd6377e2d8969dc9f93af297f5243bd)

> **System Information** - [https://gist.github.com/TheDunkShot/81af63ddb80d7b00b0ffa5ef2e31f2a3](https://gist.github.com/TheDunkShot/81af63ddb80d7b00b0ffa5ef2e31f2a3)

EDIT: I get the same feeling as Steam Linux Runtime gives me for a wine game, also Solder downloaded when I was trying to run a WINE/Proton game really strange. << I think this is the issue for me, since steam dodges 5.13-1 and downloads the steam linux runtime soldier regardless if I'm enforcing Proton for that specific version, 5.09 works fine though.

**Comment by kiroma** [comment]
Same issue here, Warframe, Fallout: New Vegas, Metal Gear Rising: Revengeance all fail.
OpenSUSE Tumbleweed, Nvidia GTX 1650 Super
System info: [https://gist.github.com/kiroma/e87f416650333b23338fb71fdc732c69](https://gist.github.com/kiroma/e87f416650333b23338fb71fdc732c69)

**Comment by RyuzakiKK** [comment]
I was able to reproduce the "X server" error and at first glance it seems like to be related to the Steam Linux Runtime Soldier not being launched correctly.

@serebit @kturques please can you follow these steps and report back if it solves the issue?

- ~~Create a temporary directory like `/tmp/steam_test`~~
- ~~In Steam uninstall the "Steam Linux Runtime - Soldier" (this seems to be the key to solve the issue)~~
- ~~Close and re launch Steam (otherwise it will continue to fail)~~
- ~~In the game that you want to launch right click with your mouse and select "Properties"~~
- ~~In "Set launch options" put this `PRESSURE_VESSEL_COPY_RUNTIME_INTO=/tmp/steam_test %command%`~~
- ~~Finally launch the game (Soldier will be automatically reinstalled and used)~~

UPDATE: With these steps I was able to solve a very similar launch issue, but it turns out that it was caused by a Steam compatibility tools file that I previously manually created. The error reported here is unlikely to be fixed by following these steps.

**Comment by kuunha** [comment]
@RyuzakiKK I tried these steps, but still have the same problem.

Cuphead [steam-268910.log](https://github.com/ValveSoftware/Proton/files/5391194/steam-268910.log)

**Comment by TheDunkShot** [comment]
Steam won't re-download Proton 5.13-1 even after a re-install of steam. For whatever reason I think Steam is trying to dodge from using proton. When proton wasn't installed/it was at 0 bytes it was going straight to using wine-staging.
EDIT: UK, Manchester server either has a corrupted proton 5.13-1 or that it doesn't download properly.
EDIT 2: Even after nuking steam, uninstalling wine, proton being downloaded and there. Still nothing, gonna try what RyuzakiKK says.

**Comment by RyuzakiKK** [comment]
@kturques Okay, thanks for trying. It seems like your issue is different from the one I experienced then.

EDIT: Are you using Wayland, right?

(I removed the request for more logging because I think I'm able to reproduce this issue under Wayland)

**Comment by ghost** [comment]
Just confirming this bug. Games tested: Rocket League, Ultra Streeg Fighter IV, a few LEGO titles, The Witcher 3: Wild Hunt, Temtem.

**Comment by alosarjos** [comment]
From the Batman Arkham Asylum bug report, I confirm this is happening in Arch Linux on Gnome+Wayland.

**Comment by kuunha** [comment]
> @kturques Okay, thanks for trying. It seems like your issue is different from the one I experienced then.
> 
> EDIT: Are you using Wayland, right?
> 
> (I removed the request for more logging because I think I'm able to reproduce this issue under Wayland)

I'm using KDE Plasma on X11

**Comment by jomarocas** [comment]
Also the same problem, games erase all data and with the game data from proton 5.0.9 no working,no driver could be loaded, all games that i test , gta v, rocket league, asassin creed origins, waiting new version of proton fix this, while i use proton 5.0.9

**Comment by DMGPage** [comment]
I can confirm. With proton 5.13-1 can't launch even a single game.

**Comment by serebit** [comment]
I'm going to edit the issue contents to be more clear on what this issue is about. Everyone who's commented, make sure your issue is characterized by THESE MESSAGES in proton logs:
```
err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
err:winediag:nodrv_CreateWindow Make sure that your X server is running and that $DISPLAY is set correctly.
```
If not, you're having a different problem.

**Comment by serebit** [comment]
> please can you follow these steps and report back if it solves the issue?

@RyuzakiKK Tried it, no dice.

**Comment by ghost** [comment]
@serebit try doing what I commented before , in your case move the intel_icd.*.json to a different directory leave  only the radeon icds and launch again any game.

**Comment by serebit** [comment]
I'm not using VK_ICD_FILENAMES, @qquique.

**Comment by ghost** [comment]
> I'm not using VK_ICD_FILENAMES, @qquique.

Thats the point, when the container runs it reads/maps the /usr/share/vulkan/icd.d and overrides with a path /overrides/vulkan/icd.d/* with all the json's and for some reason they appear in an specific order with first the igpu json (in your case the intel, mine the amdgpu) and the container sets its own env var VK_ICD_FILENAMES with all of them, and so far as I know that variable has to be set only to the icd driver that you are running the wine program (in your case the radeon in mine the nvidia) and not all the icds. So to rule out I moved the icds of the igpu to another dir so that variable in the container use the nvidia icd and it worked, in your case can be the same move the intel icds and just leave the amd just to test.

**Comment by kisak-valve** [comment]
Hello @qquique, as previously mentioned, the `VK_ICD_FILENAMES` path issue you've seen isn't related to this issue report.

@serebit, looks like @NTMan gathered a similar pattern of Proton logs in #4270. Are you also using a GNOME/Wayland or similar desktop session? (I don't know if Budgie has a wayland backend off hand)

**Comment by serebit** [comment]
@kisak-valve Budgie doesn't have a Wayland backend, so I'm using a Budgie/X11 desktop session with LightDM.

**Comment by alosarjos** [comment]
In case this helps, my Steam system information: 

https://gist.github.com/alosarjos/85fc7c25bf131d82ac64c1ab0721be72

PD: If I login in Xorg instead of Wayland, it's working fine.
PD2: Man, the Steam client also works perfectly on Xorg...Hoping to see the performance issues and stutter fixed on Wayland some day.

**Comment by tjkl1** [comment]
After testing a bit i could work around this issue most users in this thread seem to have (with an X session, not sure about wayland)

So for openSUSE Tumbleweed there are two problems to solve, for the other users maybe only one:

1. Uninstall steamtricks - This seems to rely on the old runtime and messes with "pressure vessel" - note that the steam package for now has a dependency on steamtricks, which has to be ignored

2. The "No protocol specified" error which many users seem to have - As a workaround "xhost +" does disable access control to the X Server (NOTE: Risky). After finishing you game session you can re enable it with "xhost -"

To conclude: For some reason "pressure vessel" seemingly is unable to get the right xauth cookie to make a connection to the X Server with enabled access control to the X Server.

**Comment by serebit** [comment]
`xhost +` *sort of* resolves it. When I launch DOOM Eternal with `xhost +`, the game launches in a fullscreen 640x480 mode that causes black bars to flash on the screen at a seizure-inducing pace. Changing the res back to native fixes that, though.

**Comment by sudosteve** [comment]
Are you changing the location of your Xauthority file (via XAUTHORITY env var)? I had mine set to a different location to clean up my home directory. If you run 'echo $XAUTHORITY' and see anything other than '/home/username/.Xauthority', you might be setting it somewhere (in my case it was .profile/.zprofile). Once I disabled this variable change, Proton 5.13 started working once again.

**Comment by serebit** [comment]
If Proton is looking in a certain directory for an Xauthority file, it shouldn't be. My Xauthority is in `/var/run/lightdm/serebit/xauthority` by default. I'll see if changing it to @sk8ersteve's suggestion works.

**Comment by sudosteve** [comment]
> If Proton is looking in a certain directory for an Xauthority file, it shouldn't be. 

I 100% agree and actually just opened up a new issue about this (#4296).

**Comment by sudosteve** [comment]
@serebit can you take a look at this https://github.com/ValveSoftware/steam-runtime/issues/283#issuecomment-712483139. Apparently there is a bug with pressure vessel that might be related to this problem, which they are going to push out a fix for soon. I’m just not sure if it’s really the same problem.

**Comment by serebit** [comment]
@sk8ersteve Well I gave changing the XAUTHORITY env var a shot, and it immediately caused my login to fail, so I'm pretty much stuck at the moment. Had to change it back to default.

**Comment by sudosteve** [comment]
Try it now!
https://github.com/ValveSoftware/steam-runtime/issues/283. 
https://store.steampowered.com/news/76967/
Update: "steam-runtime-tools: pick the correct DISPLAY and XAUTHORITY, fixes Gnome+Wayland and related problems (steam-runtime#283)"

**Comment by wwmm** [comment]
I have just tested it on Gnome 3.38(Arch Linux). Things improved but games are still crashing. I think the following was not present before but I do not have the old logs to verify:
```
[1024/184347.252667:ERROR:frame_sink_video_capturer_impl.cc(206)] Invalid resolutions constraints: 0x0 must not be greater than 0x0; and also within media::limits.
```
Full log file:
[gnome_3_38_crash_wayland.txt](https://github.com/ValveSoftware/Proton/files/5433819/gnome_3_38_crash_wayland.txt)

**Comment by sudosteve** [comment]
@wwmm This sounds like a separate issue. This thread is for a bug that prevents games from launching. You should maybe open up a new issue if one doesn't already exist for the problem you're facing.

**Comment by wwmm** [comment]
@sk8ersteve I had the `nodrv_CreateWindow` message described in the first post. I have updated to the latest steam client but games are still not launching. The difference is that now I see the error I mentioned before instead of the one in the first post.

**Comment by sudosteve** [comment]
@wwmm Oh, I interpreted crash as necessarily happening after launch. Nevermind.

**Comment by tjkl1** [comment]
> Try it now!
> [ValveSoftware/steam-runtime#283](https://github.com/ValveSoftware/steam-runtime/issues/283).
> https://store.steampowered.com/news/76967/
> Update: "steam-runtime-tools: pick the correct DISPLAY and XAUTHORITY, fixes Gnome+Wayland and related problems (steam-runtime#283)"

I can confirm this fixes the XAUTHORITY bug on openSUSE with atleast KDE! Thanks!

**Comment by serebit** [comment]
Yep, issue is resolved on Solus Budgie for me.

**Comment by wwmm** [comment]
I had time to do more tests. This issue was fixed. What happened is that yesterday I tested with `GreedFall` and `Atelier Ryza` and these games do not work with the latest update. But other games like `code Vein` or `A Plague Tale Innocence` are launching with no problem.

**Comment by DMGPage** [comment]
Even after latest update, I still can't play Conan Exiles with it (game just freezes on loading saved game). In same time Assassin's Creed Odyssey works just fine.

**Comment by serebit** [comment]
Seems like the issue is largely resolved, then. If no one else has issues, I'll close this sometime over the next couple days.

**Comment by serebit** [comment]
No one has piped up, so I'm closing the issue. Feel free to reopen if you're still facing this problem.

**Comment by Faithy3** [comment]
i have been unable to launch Borderlands 2 under Proton, having tried GloriousEggroll and older versions, as well as random game options, to no avail. upon creating a log it exhibited both this issue and #2878. I'm using Ubuntu Server 20.04 with GNOME 3.36.8 and X11, Nvidia 455 drivers.

**Comment by unknown321** [comment]
Same issue after upgrading Proton from 6.3-2 to 6.3-3. Unfortunately it is not possible to downgrade to lower minor version without compiling.

Specifically, update broke NieR Replicant, which was running without issues mentioned in related compability report (https://github.com/ValveSoftware/Proton/issues/4777). Proton log: [steam-1113560.log](https://github.com/ValveSoftware/Proton/files/6426368/steam-1113560.log)

Risk of Rain 2, same behaviour: [steam-632360.log](https://github.com/ValveSoftware/Proton/files/6426409/steam-632360.log)

`xhost+` doesn't help

System information https://gist.github.com/unknown321/9f325cbcaa0a4a282335ba83ed59b5fb

**Comment by mercuriete** [comment]
I am having this same error on all proton > 5.13.

First off all, I am a Gentoo user and most probably it is a error on my side but I couldn't understand what I have to do to fix it.
what I did so far:
1- installed bubblewrap with setuid
2- compiled the kernel with CONFIG_BPF_JIT=y and CONFIG_USER_NS=y

another think is that I have my game library and the proton and steam runtime soldier installed on a NTFS partition.

[sys_info.txt](https://github.com/ValveSoftware/Proton/files/6836765/sys_info.txt)

[pressure-vessel.log](https://github.com/ValveSoftware/Proton/files/6836766/pressure-vessel.log)

[steam-745920.log](https://github.com/ValveSoftware/Proton/files/6836767/steam-745920.log)

I know this is a closed issue but I don't know what to do because it seems everybody else is not having my problem.

my only workaround is to roll back to proton 5.0 and enjoy my library with an outdated proton/dxvk.

partial log:
```
5871.044:00b8:00bc:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
```

Edit:
I could fix my problem.
1- I had to delete steam runtime soldier and I had to delete libraries from NTFS partition.
2- then I had to install steam runtime soldier in my home steam library(ext4 partition)
3- then I could readd my libraries from NTFS partitions.

I hope it help someone with the same problem.
